semantic_release/history/__init__.py

```diff
@@ -114,49 +190,40 @@ def get_previous_version(version: str) -> Optional[str]:
     return get_last_version([version, "v{}".format(version)])
 
 
-def replace_version_string(content, variable, new_version):
-    """
-    Given the content of a file, find the version string and updates it.
-
-    :param content: The file contents
-    :param variable: The version variable name as a string
-    :param new_version: The new version number as a string
-    :return: A string with the updated version number
-    """
-    new_content = re.sub(
-        r'({0} ?= ?["\'])\d+\.\d+(?:\.\d+)?(["\'])'.format(variable),
-        r"\g<1>{0}\g<2>".format(new_version),
-        content,
-    )
-    # The version string did not change because above regex did not match. Use : instead of =
-    if new_content == content:
-        new_content = re.sub(
-            r'({0} ?: ?["\'])\d+\.\d+(?:\.\d+)?(["\'])'.format(variable),
-            r"\g<1>{0}\g<2>".format(new_version),
-            content,
-        )
-    return new_content
-
-
 @LoggedFunction(logger)
 def set_new_version(new_version: str) -> bool:
     """
-    Replace the version number in the correct place and write the changed file to disk.
+    Update the version number in each configured location.
 
     :param new_version: The new version number as a string.
     :return: `True` if it succeeded.
     """
-    # Read the contents of the file
-    filename, variable = config.get("semantic_release", "version_variable").split(":")
-    variable = variable.strip()
-    with open(filename, mode="r") as fr:
-        content = fr.read()
-
-    # Update the version variable
-    content = replace_version_string(content, variable, new_version)
 
-    # Write the update back to the file
-    with open(filename, mode="w") as fw:
-        fw.write(content)
+    for pattern in load_version_patterns():
+        pattern.replace(new_version)
 
     return True
+
+
+def load_version_patterns() -> List[VersionPattern]:
+    """
+    Create the `VersionPattern` objects specified by the config file.
+    """
+    patterns = []
+
+    def iter_fields(x):
+        if not x: return
+        yield from x if isinstance(x, list) else [x]
```

Comment:
Can we please change this so it supports comma separated lists as well?
```suggestion
        yield from x if isinstance(x, list) else x.split(",")
```

Reply:
Done.  I realized that I could use `csv.reader()` to make it possible to escape commas, so I did that.